src/commands/plugins.rs

```diff
@@ -242,11 +243,99 @@ impl Upgrade {
 
         if self.all {
             self.upgrade_all(manifests_dir).await
+        } else if self.name.is_none()
+            && self.local_manifest_src.is_none()
+            && self.remote_manifest_src.is_none()
+        {
+            // Default behavior (multiselect)
+            self.upgrade_multiselect().await
         } else {
             self.upgrade_one().await
         }
     }
 
+    // Multiselect plugin upgrade experience
+    async fn upgrade_multiselect(self) -> Result<()> {
+        let catalogue_plugins = list_catalogue_plugins().await?;
+        let installed_plugins = list_installed_plugins()?;
+
+        let installed_in_catalogue: Vec<_> = installed_plugins
+            .into_iter()
+            .filter(|installed| {
+                catalogue_plugins.iter().any(|catalogue| {
+                    installed.manifest == catalogue.manifest
+                        && matches!(catalogue.compatibility, PluginCompatibility::Compatible)
+                })
+            })
+            .collect();
+
+        let mut eligible_plugins = Vec::new();
+
+        // Getting only eligible plugins to upgrade
+        for installed_plugin in installed_in_catalogue {
+            let manager = PluginManager::try_default()?;
+            let manifest_location = ManifestLocation::PluginsRepository(PluginLookup::new(
+                &installed_plugin.name,
+                None,
+            ));
+
+            // Attempt to get the manifest to check eligibility to upgrade
+            if let Ok(manifest) = manager
+                .get_manifest(&manifest_location, false, SPIN_VERSION)
+                .await
+            {
+                if installed_plugin.version != manifest.version() {
+                    eligible_plugins.push((installed_plugin, manifest));
+                }
+            }
+        }
+
+        if eligible_plugins.is_empty() {
+            eprintln!("No plugins found to upgrade");
```

Comment:
This message would fit the case where the user has no plugins installed.  But by this point the list could be empty because all their plugins are up to date.  My suggestion would be to add a check for "no plugins installed" earlier in the code, and then change this to 
```suggestion
            eprintln!("All plugins are up to date");
```